python/py-black/Portfile

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           select 1.0
+
+name                py-black
+version             20.8b1
+revision            0
+
+platforms           darwin
+supported_archs     noarch
+license             MIT
+maintainers         {danchr @danchr} openmaintainer
+
+description         The uncompromising code formatter for Python.
+long_description \
+    By using Black, you agree to cede control over minutiae of \
+    hand-formatting your Python source code. In return, Black gives \
+    you speed, determinism, and freedom from pycodestyle nagging about \
+    formatting. You will save time and mental energy for more \
+    important matters. \
+    \
+    Black makes code review faster by producing the smallest diffs \
+    possible. Blackened code looks the same regardless of the project \
+    you’re reading. Formatting becomes transparent after a while and \
+    you can focus on the content instead.
+
+homepage            https://${name}.readthedocs.io/
+
+checksums           rmd160  63de27671405a15165325731e0b74e4bfe1be80d \
+                    sha256  1c02557aa099101b9d21496f8a914e9ed2222ef70336404eeeac8edba836fbea \
+                    size    1096433
+
+python.versions     36 37 38
+
+if {${subport} ne ${name}} {
+    patchfiles      patch-tests.diff
+
+    depends_build-append \
+                    port:py${python.version}-setuptools \
+                    port:py${python.version}-setuptools_scm
+
+    depends_lib-append \
+                    port:py${python.version}-appdirs \
+                    port:py${python.version}-click \
+                    port:py${python.version}-mypy_extensions \
+                    port:py${python.version}-pathspec \
+                    port:py${python.version}-regex \
+                    port:py${python.version}-toml \
+                    port:py${python.version}-typed-ast \
+                    port:py${python.version}-typing_extensions
+
+    if {${python.version} == 36} {
+        depends_lib-append \
+                    port:py${python.version}-dataclasses
+    }
+
+    depends_run-append \
+                    port:black_select
+
+    test.run        yes
+
+    select.group    black
+    select.file     ${filespath}/black-${python.version}
```

Comment:
I minor suggestion: Perhaps just you `${subport}` here? Some ports use that, others use `${python.rootname}${python.version}` and this is yet another variant…

Reply:
I would be fine with `${python.rootname}${python.version}`, but please note that the current line has the exact same result. `${subport}` is only used by one port and the result would be: ```port select --set black py38-black```, to me ```port select --set black black-38``` makes more sense. But since I made you maintainer you can pick - let me know what you want me to do ;)

Reply:
I think the difference is the hyphen. It's not consistent, but most of mine have no hyphen. (cython, mypy, pylint, pytest, etc)

Reply:
okay, I'll remove the hyphen

Reply:
I gathered a bit of statistics [as part of an old PR here](https://github.com/macports/macports-ports/pull/8524#discussion_r495879285):

```
  13 ${python.portprefix}-${python.rootname}
   1 ${python.portprefix}-py${python.rootname}
   1 ${python.portprefix}-pyls
  17 ${python.rootname}${python.version}
   5 ${python.rootname}-${python.version}
   1 ${python.rootname}tests${python.version}
   1 ${select.group}${python.version}
   1 ${subport}
   3 something${python.version}
   1 something-${python.portprefix}
   2 something-${python.version}
   1 something-print${python.version}
```

(`{python.portprefix}` is something I introduced in that PR, and it's `py${python.version}`.)

Given that `${python.portprefix}-${python.rootname}` is equivalent to `${subport}`, I'd say the two valid options are either `${subport}` or `${python.rootname}${python.version}`. My preferences would be the former, as I always found the latter a bit odd, given that commands actually use `${python.rootname}-${python.branch}`. But as long as it's either of the two, it's fine 🙂